lib/ansible/modules/packaging/os/rpm_key.py

```diff
@@ -77,7 +77,7 @@
 def is_pubkey(string):
     """Verifies if string is a pubkey"""
     pgp_regex = ".*?(-----BEGIN PGP PUBLIC KEY BLOCK-----.*?-----END PGP PUBLIC KEY BLOCK-----).*"
-    return re.match(pgp_regex, string, re.DOTALL)
+    return re.match(pgp_regex, string.decode('utf-8'), re.DOTALL)
```

Comment:
What we want to do here is avoid mixing byte and text strings.  pgp_regex is being used to try to match with string.  On python2, pgp_regex is a byte string.  On python3, pgp_regex is a text string. string.decode() will always transform the string into a text string so we end up mixing text and bytes on python2.

A better way to do this is to use:
``` python
from ansible.module_utils._text import to_native
[...]
    return bool(re.match(pgp_regex, to_native(string, errors='surrogate_or_strict'), re.DOTALL))
```

* to_native will return the string type that an unadorned literal byte string returns.
* We tell it to use the 'surrogate_or_strict' error handler so that on python3 we'll get surrogate escaped values if there are non-utf8 characters and on python2, we'll get a traceback (best we can do here, unfortunately).
* We use bool() on the return value as that prevents the function from leaking a byte-oriented regex match versus a text-oriented regex match on different versions of python.  Callers currently care only about truthiness so adding bool just ensures that they don't start depending on something undocumented.